homeassistant/components/squeezebox/__init__.py

```diff
@@ -119,16 +130,33 @@ async def async_setup_entry(hass: HomeAssistant, entry: SqueezeboxConfigEntry) -
 
     coordinator = LMSStatusDataUpdateCoordinator(hass, lms)
 
+    entry.async_on_unload(entry.add_update_listener(options_update_listener))
+
     entry.runtime_data = SqueezeboxData(
         coordinator=coordinator,
         server=lms,
+        options=SqueezeboxOptions(
+            browse_limit=entry.options.get(CONF_BROWSE_LIMIT, 1000),
+            volume_step=entry.options.get(CONF_VOLUME_STEP, 5),
+        ),
     )
 
     await coordinator.async_config_entry_first_refresh()
     await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
     return True
 
 
+async def options_update_listener(
+    hass: HomeAssistant, entry: SqueezeboxConfigEntry
+) -> None:
+    """Handle options update."""
+
+    entry.runtime_data.options.browse_limit = entry.options.get(CONF_BROWSE_LIMIT, 1000)
+    entry.runtime_data.options.volume_step = entry.options.get(CONF_VOLUME_STEP, 1000)
```

Comment:
@peteS-UK can we but these values as a const named value.

Reply:
entry.runtime_data.options.volume_step = entry.options.get(CONF_VOLUME_STEP, 1000) -> entry.runtime_data.options.volume_step = entry.options.get(CONF_VOLUME_STEP, 5)?

Reply:
Yes, can do.  These are really just if something's gone awry, since the values are defaulted and required in the configflow, but as you say, t makes sense.